backends/opt/ceed-opt-tensor.c

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+#include <ceed/ceed.h>
+#include <ceed/backend.h>
+#include "ceed-opt.h"
+
+//------------------------------------------------------------------------------
+// Tensor Contract Blocked
+//------------------------------------------------------------------------------
+int CeedTensorContractApply_Blocked_Opt(CeedTensorContract contract, CeedInt A,
```

Comment:
Interestingly, this seems to run worse with `mpiexec ./build/petsc-bps -ceed /cpu/self/opt/blocked`, with a peak near `DoFs/Sec in CG: 12.2369 (12.2369) million` vs `DoFs/Sec in CG: 16.7052 (16.7052) million` for the `ref` tensor

The SVE performance in previous commits seems similar to this unrolling

Reply:
Specializing on `t_mode` doesn't seem to have much of an effect